EESSI tutorial @ ISC'26 (#38)
EESSI/isc-tutorial: isc26-EESSI/description.tex

> %\comment{Upload detailed description of the tutorial content (2.5 pages maximum)

Reviewer: Are these limits accurate?

Author: "A detailed description of the tutorial content (Maximum pages: three)"

current versions fits that, with figure and trimming out some details (logistics part is separate)

%The main goal of this tutorial is to facilitate the daily work of its attendees.
% It will introduce EasyBuild as a tool for providing optimised, reproducible, multi-platform scientific software installations in a consistent, efficient, and user-friendly manner. We will explain the scope of EasyBuild and show attendees how to get started,
% how to tap into some advanced features,
% showcase its use on some large-scale systems,
% and show how to engage with the EasyBuild community.

The \textbf{European Environment for Scientific Software Installations
(EESSI)}\footnote{\href{https://eessi.io}{https://eessi.io}} project is a collaborative project
between different partners in the HPC community supported by the MultiXscale EuroHPC Centre of Excellence to build a
common stack of optimised scientific software installations for everything from laptops to big HPC systems and cloud
infrastructures. The project uses EasyBuild to install software on top of a compatibility layer, and uses CernVM-FS to

> distribute the software installations to client systems.

Reviewer: ```suggestion
The \textbf{European Environment for Scientific Software Installations
(EESSI)}\footnote{\href{https://eessi.io}{https://eessi.io}} project is a collaborative project
between different partners in the HPC community supported by the EuroHPC Federation Platform adn the MultiXscale EuroHPC Centre of Excellence to build a
common stack of optimised scientific software installations for everything from laptops to big HPC systems and cloud
infrastructures. The project uses EasyBuild to install software on top of a compatibility layer, and uses CernVM-FS to
distribute the software installations to client systems.

Application developers, HPC sites, and end users 
spend significant amounts of time on optimised software installations. Surveys conducted at the
\emph{``Getting Scientific Software Installed''} Birds-of-a-Feather sessions that we (co-)organised at both SC and ISC
reveal that this (still) involves a significant amount of `manual' effort.
% (and thus time/manpower).
In the SC'19 survey,
%In the most recent survey (SC'19),
less than half of the respondents consistently automate software installation,
and only ~25\% automate environment module file generation.
%and only ~25\% automatically generate environment module files consistently, as opposed to composing them manually.
%Although the problems that arise with installing scientific software are ubiquitous,
Despite these ubiquitous problems,
there is still inadequate collaboration
between HPC sites to date: less than 30\% of respondents indicated that they
work together with other HPC sites regarding software installation, even in most recent surveys.
%Since EasyBuild can help relieve these burdens and foster collaboration, a tutorial introducing this tool is highly relevant to ISC'22 attendees.
EESSI goes beyond giving recipes for how to install scientific software, it provides the installation themselves in real time. Hence, an EESSI tutorial is very relevant to ISC'26 attendees as this tool helps relieve software installation burdens and fosters
collaboration by providing an ecosystem where individual contributions benefit all.
% KH: updated to use survey results from SC19 (Tue Feb 11th 17:43 CET)
%\comment{MG: Are these numbers still OK? I seem to remember that at SC'19 more people said that they were using Spack or EB.}
%\comment{KH: 25\% Spack, 15\% EasyBuild, see slide 4 in \url{http://easybuilders.github.io/easybuild/files/SC19_GSSI_BoF/GSSI_SC19_Survey_results.pdf}; w.r.t. collaboration, it's about 33\% if you add "yes" and "sometimes" (see slide 10)}

```

Author: I've thoroughly revised the description in 3e3cd85755fbc1e149e9f115c9a9af70db71e378, please check again

EasyBuild is used by well over 100 HPC sites worldwide (incl. JSC, CSCS, the Digital Research Alliance of Canada, LUMI,
\ldots),
has over 125 unique contributors on a yearly basis,
and an active Slack channel with over 830 members.
It has reached a critical mass with a welcoming and active community guaranteeing continued development and support, and
a growing user community.
%It has reached critical mass, a point where a large user base means more features and software being added regularly, with an accelerated adoption rate, and a welcoming and engaging community.
%Attendance at the yearly EasyBuild User Meetings has been growing steadily, with close to 100 attendees at the 7th (virtual) edition in Jan'22.
%As such, we strongly believe that EasyBuild, and the community behind it, will have a positive impact on the attendees' work.
% The tutorial content is designed to help with getting started quickly and benefit from EasyBuild right after the conference.

> %The tutorial content will be designed to help the attendees getting started with the tool, so they can immediately benefit from it after the conference.

Reviewer: ```suggestion
```

Author: I've thoroughly revised the description in 3e3cd85755fbc1e149e9f115c9a9af70db71e378, please check again

EESSI is an initiative built on the foundations of the EasyBuild community, and goes one step further by providing a

> truly uniform software stack. It already has a wide spectrum of collaborators despite being a relatively young project.

Reviewer: This is probably the place to mention EFP and the award

Author: That's a TODO, I'll look into that

EESSI is motivated by the observation that the landscape of computational science is changing in various ways.

> Additional families of general-purpose microprocessors including Arm 64-bit (aarch64) and RISC-V on top of th

Reviewer: ```suggestion
Additional families of general-purpose microprocessors including Arm 64-bit (aarch64) and RISC-V on top of the
```

Author: fixed in 3e3cd85755fbc1e149e9f115c9a9af70db71e378

increasing the diversity in system architectures. The rapid expansion of computational science beyond traditional
domains like physics and computational chemistry, including bioinformatis, Machine Learning (ML) and Artificial
Intelligence (AI), etc. leads to a significant growth of the software stack that is used for running scientific
workloads. The emergence of commercial cloud infrastructure (Amazon EC2, Microsoft Azure, ...) and private cloud
infrastructure (OpenStack) has competitive advantages over on-premise infrastructure for computational workloads, such
as near-instant availability, increased flexibility, a broader variety of hardware platforms, and faster access to new
generations of microprocessors. In addition the manpower that is available in the HPC user support teams that are
responsible for helping scientists with running the software they require on high-end (and complex) infrastructure like
supercomputers (and beyond) is limited. These reasons indicate that there is a strong need for more collaboration on
building and installing scientific software to avoid duplicate work across computational scientists and HPC user support
teams.

> The main goal of EESSI is to provide a collection of scientific software installations that work across a wide range of
> different platforms, including HPC clusters, cloud infrastructure, and personal workstations and laptops, without making
> compromises on the performance of that software.

Reviewer: ```suggestion
The main goal of EESSI is to provide a collection of scientific software installations that work across the computing
continuum, including HPC clusters, cloud infrastructure, and personal workstations and laptops, without making
compromises on the performance of that software.
```

Author: I've thoroughly revised the description in 3e3cd85755fbc1e149e9f115c9a9af70db71e378, please check again

The tutorial covers several critical aspects of EESSI, including its motivation, goals, and high-level design, as
demonstrated through hands-on demonstrations and guided illustrations. The main features and functionalities of EESSI
are highlighted, as well as the different ways to access the software stack, such as native installation, use in an HPC
production setup, or without administrator privileges. The tutorial also covers various use cases for EESSI, including
portable workflows, integration into a CI environment, adding software to the EESSI, and also configuring GPU support.
%Troubleshooting access, testing the software, evaluating the performance of EESSI and configuring GPU support are also covered.

> Extensive instructor-led examples are used to put the theoretical knowledge into practice.

Reviewer: ```suggestion
The tutorial covers several critical aspects of EESSI, including its motivation, goals, and high-level design. The main features and functionalities of EESSI
are highlighted, and
demonstrated through hands-on material with guided challenges. The tutorial also covers various use cases for EESSI, including
portable workflows, integration into a CI environment, adding software to the EESSI, and advanced use cases.
%Troubleshooting access, testing the software, evaluating the performance of EESSI and configuring GPU support are also covered.
Extensive instructor-led examples are used to put the theoretical knowledge into practice.
```

Author: I've thoroughly revised the description in 3e3cd85755fbc1e149e9f115c9a9af70db71e378, please check again
